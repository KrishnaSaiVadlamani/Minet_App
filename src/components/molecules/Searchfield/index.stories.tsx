import { ComponentStory, ComponentMeta } from '@storybook/react'
import React from 'react'
import SearchField from './index'

export default {
  title: 'Molecules/SearchField',
  component: SearchField,
} as ComponentMeta<typeof SearchField>

const Template: ComponentStory<typeof SearchField> = (args) => (
  <SearchField {...args} />
)

export const SearchFieldWithoutFilter = Template.bind({})

SearchFieldWithoutFilter.args = {
  placeholder: 'Search all assets',
  filter: false,
}

export const SearchFieldWithFilter = Template.bind({})

SearchFieldWithFilter.args = {
  placeholder: 'Search all assets',
  filter: true,
}