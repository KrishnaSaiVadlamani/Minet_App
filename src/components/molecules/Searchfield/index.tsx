import { TextField, styled, IconButton, Stack, Divider } from '@mui/material'
import React from 'react'
import theme from '../../../theme'
import IconComponent from '../../atoms/Icons'
import search from '../../../../public/assets/icons/search.svg'
import filter from '../../../../public/assets/icons/filter.svg'

interface SearchFieldProps {
  placeholder: string
  filter: boolean
  handleChange: (value: string) => void
  handleClick?: React.MouseEventHandler<HTMLButtonElement>
  backgroundColor?: string
  value?: string
}

const StyledSearchField = styled(TextField)((props: SearchFieldProps) => ({
  '& .MuiOutlinedInput-root': {
    width: props.filter ? '348px' : '230px',
    height: '40px',
    paddingRight: '5px',
    borderRadius: '4px',
    backgroundColor: props.backgroundColor,
    '&:hover fieldset': {
      border: `1px solid ${theme.palette.greyColors.grey100}`,
    },
  },
  '& .MuiOutlinedInput-notchedOutline': {
    border: `1px solid ${theme.palette.greyColors.grey100} !important`,
    padding: '0px',
  },
  'input': {
    '&::placeholder': {
      color: theme.palette.textColor.main,
      fontFace: theme.typography.body2,
    }
  }
}))

const StyledIconButton = styled(IconButton)({
  paddingTop: '13px',
  '&:hover': {
    background: 'none',
  }
})

const StyledDivider = styled(Divider)({
  height: '30px', 
})

const renderStack = (handleClick: React.MouseEventHandler<HTMLButtonElement>) => {
  return (
    <Stack
      direction="row"
      divider={<StyledDivider orientation="vertical" variant = 'middle' flexItem style={{marginTop: '10px'}}/>}
      spacing={1}
    >
      {renderSearchIcon(handleClick)}
      {renderFilterIcon()}
    </Stack>
  )
}

const renderSearchIcon = (handleClick: React.MouseEventHandler<HTMLButtonElement>) => {
  return (
    <StyledIconButton onClick={handleClick}>
        <IconComponent src={search} height="20.31px" width="20.31px" />
      </StyledIconButton>
  )
}

const renderFilterIcon = () => {
  return (
    <StyledIconButton >
        <IconComponent src={filter} height="22px" width="18px"/>
      </StyledIconButton>
  )
}
const renderEndAdornment = (
  filter: boolean,
  handleClick: React.MouseEventHandler<HTMLButtonElement>
) => {
  if (filter) {
    return (renderStack(handleClick))
  } else {
    return (
      renderSearchIcon(handleClick)
    )
  }
}

const SearchField: React.FC<SearchFieldProps> = ({
  placeholder,
  handleChange,
  handleClick,
  filter,
  backgroundColor,
  value
}) => {
  return (
    <StyledSearchField
    variant='outlined'
      placeholder={placeholder}
      onChange={(event) => handleChange(event.target.value)}
      handleChange={handleChange}
      handleClick={handleClick}
      filter={filter}
      InputProps={{
        endAdornment: renderEndAdornment(filter, handleClick ? handleClick : () => {}),
      }}
      backgroundColor={backgroundColor}
      value={value}
    />
  )
}

export default SearchField
