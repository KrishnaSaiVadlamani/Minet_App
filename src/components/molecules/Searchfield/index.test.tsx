import renderer from 'react-test-renderer';
import Searchfield from '.';
import {render, screen} from '@testing-library/react';
import '@testing-library/jest-dom'
import React from 'react';

it('renders search bar correctly', () => {
    const tree = renderer
      .create(<Searchfield filter={false} placeholder={'Search all assets'} handleClick = {() => {}} handleChange={() => {}}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders search bar with filter correctly', () => {
    const tree = renderer
      .create(<Searchfield filter={true} placeholder={'Search all assets'} handleClick = {() => {}} handleChange={() => {}}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
});

describe('Molecules/Searchfield', () => {
  test('Renders search bar without filter', () => {
      render(<Searchfield filter={false} placeholder={'Search all assets'} handleClick = {() => {}} handleChange={() => {}}/>)
      const searchBar = screen.getByTestId('search-bar')
      expect(searchBar).toBeInTheDocument();
      const button = screen.getByRole('button');
      expect(button).toBeInTheDocument();
  })
  test('Renders search bar with filter', () => {
      render(<Searchfield filter={true} placeholder={'Search all assets'} handleClick = {() => {}} handleChange={() => {}}/>)
      const searchBar = screen.getByTestId('search-bar')
      expect(searchBar).toBeInTheDocument();
      const button = screen.getAllByRole('button');
      expect(button).toHaveLength(2);
  })
})