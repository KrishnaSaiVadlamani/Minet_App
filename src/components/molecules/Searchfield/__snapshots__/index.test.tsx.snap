// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders search bar correctly 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root css-jr539y-MuiFormControl-root-MuiTextField-root"
  data-testid="search-bar"
  filter={false}
  handleChange={[Function]}
  handleClick={[Function]}
>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
      disabled={false}
      id=":r0:"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Search all assets"
      required={false}
      type="text"
    />
    <button
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-h51xkb-MuiButtonBase-root-MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <div>
        <img
          alt="icon"
          className="css-14oiz6v"
          data-testid="iconComponent"
          height="20.31px"
          src={
            {
              "process": [Function],
            }
          }
          width="20.31px"
        />
      </div>
    </button>
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-ihdtdm"
      >
        <span
          className="notranslate"
        >
          ​
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`renders search bar with filter correctly 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root css-jbv63a-MuiFormControl-root-MuiTextField-root"
  data-testid="search-bar"
  filter={true}
  handleChange={[Function]}
  handleClick={[Function]}
>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
      disabled={false}
      id=":r1:"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Search all assets"
      required={false}
      type="text"
    />
    <div
      className="css-w4z10b-MuiStack-root"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-h51xkb-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <div>
          <img
            alt="icon"
            className="css-14oiz6v"
            data-testid="iconComponent"
            height="20.31px"
            src={
              {
                "process": [Function],
              }
            }
            width="20.31px"
          />
        </div>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <hr
        className="MuiDivider-root MuiDivider-middle MuiDivider-vertical MuiDivider-flexItem css-eg8219-MuiDivider-root"
        style={
          {
            "marginTop": "10px",
          }
        }
      />
      <button
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-h51xkb-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <div>
          <img
            alt="icon"
            className="css-mw0g01"
            data-testid="iconComponent"
            height="22px"
            src={
              {
                "process": [Function],
              }
            }
            width="18px"
          />
        </div>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-ihdtdm"
      >
        <span
          className="notranslate"
        >
          ​
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
